tests/end-to-end/backup/pkg/tests/servicecatalog/app_broker.go

```diff
@@ -211,6 +230,58 @@ func (f *appBrokerFlow) createApplication() error {
 	return err
 }
 
+// createChannel creates a Knative channel that would normally be installed by the application controller upon helm-chart installation. As we skip the installation part,
+// we need to ensure that the channel exists as it is required by the knative broker.
+func (f *appBrokerFlow) createChannel() error {
+	channel := &messagingv1alpha1.Channel{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Channel",
+			APIVersion: "messaging.knative.dev/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: applicationName,
+			Labels: map[string]string{
+				"application-name": applicationName,
+			},
+		},
+	}
+
+	_, err := f.messagingInterface.Channels(integrationNamespace).Create(channel)
+	if errors.IsAlreadyExists(err) {
+		return nil
+	}
+	return err
+}
+
+func (f *appBrokerFlow) waitForChannel() error {
```

Comment:
why are we waiting for messagingchannel but not the httpsource?

Reply:
In this case we do not send an event. so the only thing we are waiting for should be the channel.